### PR TITLE
[logfile]: Add handling of Sairedis rec filename 

### PR DESCRIPTION
**Pull request by arlakshm**
The PR implements the handling of the sairedis rec filename passed to the orchagent.
Coresponding swss pr : https://github.com/Azure/sonic-swss/pull/1546
This change is mainly to be used for multi asic devices where each asic.

### CODE REVIEW

**Comment by kcudnik** [comment]
this will also needs to be handled by sonic syslog rotate, since how it's hardcoded to sairedis.rec i think

**Comment by arlakshm** [comment]
> this will also needs to be handled by sonic syslog rotate, since how it's hardcoded to sairedis.rec i think

Yes, I will create a PR in sonic-buildimage for logrotate

**Comment by lguohan** [comment]
@kcudnik  to approve

File: lib/inc/sairedis.h [hunk 1]
```diff
@@ -194,4 +194,15 @@ typedef enum _sai_redis_switch_attr_t
...
     SAI_REDIS_SWITCH_ATTR_CONTEXT,
 
+    /**
+     * @brief Recording log filename.
+     *
+     * Default valus is sairedis.rec
+     *
+     * @type sai_s8_list_t
+     * @flags CREATE_AND_SET
+     * @default empty
```
Comment (kcudnik) [inline lib/inc/sairedis.h:204]: you could make this @default "sairedis.rec"

File: lib/src/Recorder.cpp [hunk 2]
```diff
@@ -97,6 +97,58 @@ bool Recorder::setRecordingOutputDirectory(
...
 }
 
+bool Recorder::setRecordingFilename(
+    _In_ const sai_attribute_t &attr)
+{
+    SWSS_LOG_ENTER();
+
+    if (attr.value.s8list.count == 0)
+    {
+        m_recordingFileName = "sairedis.rec";
```
Comment (kcudnik) [inline lib/src/Recorder.cpp:107]: this name is also used in constructor, can you make a define on top of file and reuse define here ?

**Review by kcudnik — CHANGES_REQUESTED** [review]

File: lib/src/Recorder.cpp (hunk 2 above)
Comment (arlakshm) [reply lib/src/Recorder.cpp]: address the commit in the latest commit

File: lib/inc/sairedis.h (hunk 1 above)
Comment (arlakshm) [reply lib/inc/sairedis.h]: address the commit in the latest commit

**Review by kcudnik — APPROVED** [review]

**Pull request merged**